tests/Feature/ManageBillingsTest.php

```diff
@@ -31,17 +34,22 @@ public function guest_cannot_manage_billings()
     public function a_user_can_create_billing()
     {
         $this->signIn();
-
         $this->get(route('billings.create'))->assertStatus(200);
+
+        $file = UploadedFile::fake()->create('billing.csv');
+
         $attributes = [
             'name' => $this->faker->sentence(4),
             'working_days_rate' => $this->faker->randomFloat(4, 0, 1),
             'saturday_rate' => $this->faker->randomFloat(4, 0, 1),
+            'import_file' => $file
         ];
 
         $this->post(route('billings.store'), $attributes)
             ->assertRedirect(route('billings.index'))
             ->assertSessionHas('success');
+
+        unset($attributes['import_file']);
         $this->assertDatabaseHas('billings', $attributes);
```

Comment:
To jest w pamięci czy jakaś testowa baza danych?

Reply:
Laravel ma klasę Illuminate\Foundation\Testing\RefreshDatabase
która jest odpowiedzialna za usuwanie wszystkich rzeczy, które były dodane do bazy podczas testu.
trzeba tylko użyć jej w klasie testu i samo się ogarnia